operator/iamkeycontroller/update.go

```diff
@@ -5,13 +5,22 @@ import (
 
 	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
 	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	exov1 "github.com/vshn/provider-exoscale/apis/exoscale/v1"
 	controllerruntime "sigs.k8s.io/controller-runtime"
 )
 
 // Update implements managed.ExternalClient.
 // exoscale.com does not allow any updates on IAM keys.
 func (p *IAMKeyPipeline) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
 	log := controllerruntime.LoggerFrom(ctx)
-	log.V(1).Info("Updating resource (noop)")
-	return managed.ExternalUpdate{}, nil
+	log.V(1).Info("Updating role")
+
+	iamKey := fromManaged(mg)
+	iamKey.SetConditions(exov1.Updating())
+
+	role := createRole(iamKey.Spec.ForProvider.KeyName, iamKey.Spec.ForProvider.Services.SOS.Buckets)
```

Comment:
Why do you create the role in `Update()` method?

Reply:
`createRole()` creates the role object with the policy, it doesn't create anything on the API.

Should I rename this function?